mlir/include/mlir/Analysis/Presburger/Matrix.h

```diff
@@ -239,6 +236,36 @@ class IntMatrix : public Matrix<MPInt> {
   /// Divide the columns of the specified row by their GCD.
   /// Returns the GCD of the columns of the specified row.
   MPInt normalizeRow(unsigned row);
+
+  // Return the integer inverse of the matrix, leaving the calling object
+  // unmodified.
+  std::optional<IntMatrix> integerInverse();
+
+  // Compute the determinant of the matrix by converting it to row echelon
+  // form and then taking the product of the diagonal.
```

Comment:
the algorithm doesn't matter here. the header level documentation is for the user to decide whether to use the function. Just mention a complexity estimate here.